gplugins/utils/get_component_with_net_layers.py

```diff
@@ -0,0 +1,105 @@
+import copy
+
+import gdsfactory as gf
+from gdsfactory.typings import Component, LayerStack
+
+
+def remove_empty_layerstack_layers(
+    component: Component,
+    layerstack: LayerStack,
+) -> LayerStack:
+    """Returns a new layerstack without layers that don't appear in the provided component.
+
+    Arguments:
+        component: to process.
+        layerstack: to process.
+
+    Returns:
+        new_layerstack: without layers that do not appear in component
+    """
+    new_layerstack = layerstack.copy()
+
+    layers_present = component.layers
+    layernames_dict = new_layerstack.get_layer_to_layername()
+    layernames_present = [
+        name
+        for sublist in [layernames_dict[layer] for layer in layers_present]
+        for name in sublist
+    ]
+    for key in list(new_layerstack.layers.keys()):
+        if key not in layernames_present:
+            del new_layerstack.layers[key]
+
+    return new_layerstack
+
+
+def get_component_with_net_layers(
+    component: Component,
+    layerstack: LayerStack,
+    portnames: list[str],
```

Comment:
rename to `port_names`

Reply:
addressed

Reply:
did you push?
i don't see it addressed

Reply:
Pushed now. Should I squash before we merge?